mesonbuild/interpreter/interpreter.py

```diff
@@ -2801,6 +2801,12 @@ def extract_incdirs(self, kwargs, key: str = 'include_directories') -> T.List[bu
             if isinstance(p, build.IncludeDirs):
                 result.append(p)
             elif isinstance(p, str):
+                if key == 'd_import_dirs' and os.path.normpath(p).startswith(self.environment.get_source_dir()):
+                    FeatureDeprecated.single_use('Building absolute path to source dir is not supported',
+                                                 '1.6.0', self.subproject,
+                                                 'Use a relative path instead. This will become a hard error in the future.',
```

Comment:
We don't need the hard-error bit, deprecation implies that.

```suggestion
                                                 'Use a relative path instead.',
```

Reply:
Done. I had used the old error message verbatim but I admit that I found it a little bit user hostile. Your suggestion is much better.